zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Collection.java

```diff
@@ -285,6 +301,36 @@ public boolean deleteCollection(HttpServletRequest request, HttpServletResponse
         return true;
     }
 
+    private void deleteCollection(com.github.onsdigital.zebedee.model.Collection c, Session s)
+            throws InternalServerError, NotFoundException, BadRequestException {
+        // Delete the collection.
+        try {
+            collections.delete(c, s);
+        } catch (Exception ex) {
+            String message = format("error attempting to delete collection: {0}", c.getId());
+            throw new InternalServerError(message, ex);
+        }
+
+        // Cancel any scheduled publish for this collection.
+        scheduleCanceller.cancel(c);
+
+        // Remove the collection encryption key from the keyring
+        User user = null;
+        try {
+            user = KeyringUtil.getUser(usersService, s.getEmail());
+        } catch (Exception ex) {
+            String message = format("error attempting to get user from session details: {0}", s.getEmail());
+            throw new InternalServerError(message, ex);
```

Comment:
when we use the try catch block for get user all the exception get's caught as internal server error. In this case do we still need NotFoundException and BadRequestException defined in the function signature?